sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -888,7 +884,6 @@ private void open(String uId,
       LOG.debug("Opening {}.", filename);
       final WritableByteChannelFactory factory =
           getWriteOperation().getSink().writableByteChannelFactory;
-      mimeType = factory.getMimeType();
```

Comment:
maybe we should let factories declare an optional mime type here?

it looks to me like this whole support was not really thought out.